addons/html_editor/static/src/main/media/media_dialog/video_selector.js

```diff
@@ -129,22 +129,7 @@ export class VideoSelector extends Component {
             }
         });
 
-        onMounted(async () => {
-            await Promise.all(
-                this.props.vimeoPreviewIds.map(async (videoId) => {
-                    const { thumbnail_url: thumbnailSrc } = await this.http.get(
-                        `https://vimeo.com/api/oembed.json?url=http%3A//vimeo.com/${encodeURIComponent(
-                            videoId
-                        )}`
-                    );
-                    this.state.vimeoPreviews.push({
-                        id: videoId,
-                        thumbnailSrc,
-                        src: `https://player.vimeo.com/video/${encodeURIComponent(videoId)}`,
-                    });
-                })
-            );
-        });
+        onMounted(async () => this.prepareVimeoPreviews());
```

Comment:
This should be done during the onWillStart. Currently we have 2 renderings caused by 1 initial and one caused by prepareVimeoPreviews. onMounted does not wait for the async function.